app/vmalert/rule/alerting.go

```diff
@@ -658,12 +658,14 @@ func pendingAlertStaleTimeSeries(ls map[string]string, timestamp int64, includeA
 		})
 	}
 	// __name__ already been dropped, no need to check duplication
-	alertsLabels := append(baseLabels, prompbmarshal.Label{Name: "__name__", Value: alertMetricName})
+	alertsLabels := append([]prompbmarshal.Label{}, baseLabels...)
```

Comment:
I think, it's better to pre-allocate slice with capacity and append elements to it. Since we already know needed size. Because the second append call with `__name__` label may cause additional allocation.

 Propose code:
```go
alertLabels := make([]prompbmarshal.Label{},len(baseLabels)+ 1)
alertLabels = append(alertLabels,baseLabels...)
alertLabels = append(alertLabels,prompbmarshal.Label{Name: "__name__", Value: alertMetricName})
```

 Helper function: `alertStateLabelsFromBase` could be usefull:

```go
func alertStateLabelsFromBase(baseLabels []prompbmarshal.Label,stateLabelValue string) [][]prompbmarshal.Label{
resultLabels := make([]prompbmarshal.Label{},len(baseLabels)+ 1)
resultLabels = append(resultLabels,baseLabels...)
resultLabels = append(resultLabels,prompbmarshal.Label{Name: "__name__", Value: alertMetricName})
return stateLabelValue
}
```

usage:

```go
alertsForStateLabels := alertStateLabelsFromBase(baseLabels,alertForStateMetricName)
alertsLabels := alertStateLabelsFromBase(baseLabels,alertMetricName)
```